examples/java/spring-boot/design/onion-architecture.md

```diff
@@ -0,0 +1,23 @@
+# Onion Architecture
+
+We have [decided to structure our application using an approach called Onion Architecture](../../../../adr/0005-onion-architecture.md). 
+
+The classes in this reference implementation are split into four packages, each of which represents a layer in the Onion Architecture model:
+- infrastructure
+- application
+- domain.service
+- domain.model
+
+![here](artefacts/onion-architecture.drawio.svg)
+
+Anything can depend on something from an inner layer, and inner layers don't know anything about outer layers.
+
+This allows us to think about the domain and business logic separately from how we implement the infrastructure layer.
+
+Below is a class diagram generated from our codebase, showing how the classes in each package relate to each other:
+
+![PlantUML Diagram for onion architecture](https://www.plantuml.com/plantuml/png/xLTVJyCs47zkFaKUMmqiyLgXYWoc3Gq60j6NQQfCScrD7JlpEPqGuxlFJjCw9W5IwabeceHeV7_EFzzTVZcVeQPAbpcFSCL4GHNDOrpIJFx5n2AUKuuG5JHTqGN4vOeiv9hGWX4ibD4_Igb9fYN6Bz4WvHJn0nkI5CpQ3OwFpRyVp6fehAQWa4anZk-rjJwA1St1hJB0LB52UoQbucwHIg5fgfr0gWKL3Adlaycqp47eIlpW0emC5a8i80TcEDfghRpUm1SsSFcAj_luMlYpf1o7yl5f7DyyFa6gZTfaY0CLLh-ak3oN6LWl9hGLsNpuGPe7z3LQRfvzQUcZU8qYy_TnJMbHS9PMExZIVcfAaeQRg_GBscoaht5EcHXIjS0jcD__65Lu1htH40IrPYbK7UTmzNIoBNY_UQxyAu06eHCF5HGIcPRgUHmRyTrkERAMX5Js2WnyM4F2gynUCTIdndXYbAzHlmmpCLSKjIfJNIeWksqPk9TfVrVoKrxePmzten1p9h8MvA-WpMK674aEccPKKx8-eRnOKWVya2YsIETIQ2Kv1zM0xyc7TKt7iIlUKRm-IMo8QrDyKEDusbpMRtgxV3-MV7LTWxW3BAH0i6a9SfWBisYSOG4woJ9ckvBoMoKDBiq0BGo3yhHcgmqrJ6g7you2GYT_mu2ZsktYFOf2N-TJh1-YWymUxYxh82rjYsEjyJRMRoqUmrhe6D_jVFCkn-4RfK_tVfGMP2ie6y08QRAoNeBgpSlOKSbdxooc7IalOHUncw63TLfN6n_OUOdtIUpQW3YJEYbEu7tUFt4UkHsnoKzRbucBZZOkTfXyguA2AFF0zVGpLfKskRsS_NmpdJtyEflEJcR9rUqlIVkxErSodrhamnh_1cY_adRF55MWVw696_KCshkDkBrRjjjdnur6R2FRtpfE9RBc6-b_gX-zqeHkmrRe_Rjl_ncc6WgyOQk_5_4cYqQdfbAiGMdRHzMn_CFr0vf9EoXlwk3WVMxxRIvO_r-m1lqht1hz1mONg3bDz_OchS6_xeNswD-whNpj4aN8f6TZHa7kCeJyFJaiAq5m13zthwkGtJNyI145ZzZ26Nhq-OwGcomVAb1GLC7o5nHWDJqTyENj_oA6toLST_IdhtqEV0OYAtF-3m00)
```

Comment:
This will need to be updated once we've rebased `main` to make sure we have the new `CorrelationIdFilter`